internal/test_cases/history_persistence_test_case.go

```diff
@@ -0,0 +1,114 @@
+package test_cases
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/codecrafters-io/shell-tester/internal/assertions"
+	"github.com/codecrafters-io/shell-tester/internal/logged_shell_asserter"
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+// HistoryPersistenceTestCase checks that after loading history from a file, the shell's history output matches the file's contents.
+type HistoryPersistenceTestCase struct {
+	// PreviousCommands are the commands that were executed before loading the history file
+	PreviousCommands []CommandResponseTestCase
+
+	// FilePath is the path to the file containing history commands
+	FilePath string
+
+	// SuccessMessage is the message to log in case of success
+	SuccessMessage string
+
+	// Was history command executed before loading the history file
+	WasHistoryCommandExecuted bool
+
+	// ExpectHistoryRCommand controls whether to expect a history -r command in the history
+	ExpectHistoryRCommand bool
```

Comment:
The interface here seems kinda wonky - let's see if we can improve this over our call tomo